devito/ir/iet/utils.py

```diff
@@ -102,7 +99,7 @@ def derive_parameters(iet, drop_locals=False):
     free_symbols = FindSymbols('free-symbols').visit(iet)
 
     # Filter out function base symbols and use real function objects
-    function_names = [s.name for s in functions]
+    function_names = set(flatten([(s.name, s._C_name) for s in functions]))
```

Comment:
can probably use `set(*[...` rather than flatten

Reply:
fairly sure I tried but for whatever reason it wasn't working, can try again